### ISSUE TITLE
TDateTime in SO(...) paremeters

### ISSUE THREAD

**Issue by EgorovAlex** [onryldz/x-superobject#153]
Hello, 

Is it impossible to create Json TDateTime field with SO paremeter?

var X := SO(['Date', Now]);
var sJson := X.AsJSON; // Here we have Double value for 'Date' in Json

var X := SO;
X.D['Date'] := Now;
var sJson := X.AsJSON; // Here we have DateTime value, as expected

**Comment by vkrapotkin** [comment]
The JSON itself doesn't specify the format of date/time, so these fields are stored just as string (or integer or FP-number)
Actual format is choosed by a programmer.
The back conversion is not obvious by the same reason. 
But you can inherit class TJsonObject and add cusom property D or whatever else.

**Comment by EgorovAlex** [comment]
No, this example demostrate 2 method's of initialization for one library and result must be identical!
Or (if is it impossible) all restrictions must be described.

var S: string := 'Str';
var D: TDateTime := Now;
var I: Integer := 123;
var F: Double := 12.12;
var sJson1 := SO(['S', S, 'D', D, 'I', I, 'F', F]).AsJSON;
var X := SO;
X.S['S'] := S;
X.D['D'] := D;
X.I['I'] := I;
X.F['F'] := F;
var sJson2 := X.AsJSON;

sJson1 = '{"S":"Str","D":45304.7648217708,"I":123,"F":12.12}'
sJson2 = '{"S":"Str","D":"2024-01-13T18:21:20.601","I":123,"F":12.12}'

**Comment by EgorovAlex** [comment]
This can be fixed with using another declaration:
function SO(... array of TValue)
with TValue is it possible to determine TDateTime type,
but possible to reduce supported versions of Delphi (and possible Lazarus)
